src/dos/dos_files.cpp

```diff
@@ -983,26 +983,59 @@ uint8_t FCB_Parsename(uint16_t seg,uint16_t offset,uint8_t parser ,char *string,
 	return ret;
 }
 
-static void DTAExtendName(char * const name,char * const filename,char * const ext) {
-	char * find=strchr(name,'.');
-	if (find && find!=name) {
-		strcpy(ext,find+1);
-		*find=0;
-	} else ext[0]=0;
-	strcpy(filename,name);
-	size_t i;
-	for (i=strlen(name);i<8;i++) filename[i]=' ';
-	filename[8]=0;
-	for (i=strlen(ext);i<3;i++) ext[i]=' ';
-	ext[3]=0;
+// DTAExtendName takes the fullname and writes the name and extension portions
+// into those arguments, respectively, and also space-pads them up to their
+// full 8.3 lengths. Note: the space-padding is required to fully populate DOS
+// data memory structures. Also note that DOS filenames might not have a dot or
+// extension.
+static void DTAExtendName(const char fullname[DOS_NAMELENGTH_ASCII],
+                          char name[DOS_MFNLENGTH_STRING],
+                          char ext[DOS_EXTLENGTH_STRING])
+{
+	assert(fullname && fullname[0] != '\0');
+	assert(name && name[0] == '\0');
+	assert(ext && ext[0] == '\0');
```

Comment:
Hmm, we could just overwrite some existing content in `name` and `ext`, right? I don't really see why it should be disallowed.

Reply:
Also, can't we just allow it to accept empty strings? Seems a little safer perhaps?

Reply:
> Hmm, we could just overwrite some existing content in name and ext, right? I don't really see why it should be disallowed.

Unfortunately with the refactored code, it still isn't robust enough to handle the case where either the extension or first name are zero length (which causes the strncpy's to not run), in which case, the zero-padding portion of code will wind up processing the original junk content (in `name` and `ext`) and zero-padding that, instead of processing the newly written `name` and `ext` values.

> Seems a little safer perhaps?

Indeed; will do a full refactor :+1:

Reply:
Refactored.